### PR TITLE
docs: add doc for loki ingester protocol

### PR DESCRIPTION
**Pull request by paomian**
## What's Changed in this PR

* add doc for loki ingester

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/user-guide/ingest-data/for-observerbility/loki.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        "job" = "greptime"
+        "from" = "alloy"
+    }
+}
+```
+
+You can run the following command to check the data in the table:
+
+```sql
+mysql> select * from loki_demo_logs;
```
Comment (nicecui) [inline docs/user-guide/ingest-data/for-observerbility/loki.md:43]:
```suggestion
select * from loki_demo_logs;
```

File: docs/user-guide/ingest-data/for-observerbility/loki.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Data Model
+
+The Loki logs data model is mapped to the GreptimeDB data model according to the following rules:
+
+Default table schema:
+
+```sql
+mysql> desc loki_demo_logs;
```
Comment (nicecui) [inline docs/user-guide/ingest-data/for-observerbility/loki.md:90]:
```suggestion
DESC loki_demo_logs;
```

File: docs/user-guide/ingest-data/for-observerbility/loki.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ENGINE=mito
+WITH(
+  append_mode = 'true'
+)
+1 row in set (0.00 sec)
+```
+
+## Data Model
+
+The Loki logs data model is mapped to the GreptimeDB data model according to the following rules:
```
Comment (nicecui) [inline docs/user-guide/ingest-data/for-observerbility/loki.md:85]: please explain more about the rules:
- how to specify tags and time index.
- the data type conversion rules

File: docs/user-guide/ingest-data/for-observerbility/loki.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+To send OpenTelemetry Logs to GreptimeDB through OpenTelemetry SDK libraries, use the following information:
+
+* URL: `http{s}://<host>/v1/loki/api/v1/push`
+* Headers:
+  * `X-Greptime-DB-Name`: `<dbname>`
+  * `Authorization`: `Basic` authentication, which is a Base64 encoded string of `<username>:<password>`. For more information, please refer to [Authentication](https://docs.greptime.com/user-guide/deployments/authentication/static/) and [HTTP API](https://docs.greptime.com/user-guide/protocols/http#authentication).
+  * `X-Greptime-Log-Table-Name`: `<table_name>` (optional) - The table name to store the logs. If not provided, the default table name is `loki_logs`.
+
+### Example Code
+
```
Comment (nicecui) [inline docs/user-guide/ingest-data/for-observerbility/loki.md:16]: Please briefly explain alloy and the configuration file

**Review by nicecui — COMMENTED** [review]
Please:
1. Move `DESC` and `SHOW CREATE` SQLs to the data model section.
2. One SQL statement uses one code block.

File: docs/user-guide/ingest-data/for-observerbility/loki.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+| line               | String              |      | YES  |         | FIELD         |
++--------------------+---------------------+------+------+---------+---------------+
+5 rows in set (0.00 sec)
+```
+
+- greptime_timestamp: The timestamp of the log.
+- line: The log message.
+
+if you specify the external labels, we will add them as tags to the table schema. like `job` and `from` in the above example.
+We can't specify tags manually, all labels are treated as tags and string type.
```
Comment (nicecui) [inline docs/user-guide/ingest-data/for-observerbility/loki.md:83]:
```suggestion
You can't specify tags manually, all labels are treated as tags and string type.
```

**Review by nicecui — COMMENTED** [review]
what happened if developers manually created table via SQL?

**Comment by paomian** [comment]
> what happened if developers manually created table via SQL?

It will fail. because of semantic type mismatch.

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+| line               | String              |      | YES  |         | FIELD         |
++--------------------+---------------------+------+------+---------+---------------+
+5 rows in set (0.00 sec)
+```
+
+- greptime_timestamp: 日志的时间戳。
+- line: 日志消息。
+
+如果您指定了外部 label，我们会将它们添加为表结构中的 tag。例如上面的 `job` 和 `from`。
+我们不能手动指定 tag，所有 label 都被视为 tag 并且类型为字符串。请不要尝试使用 SQL 提前创建表来指定 tag 列，这会导致类型不匹配而写入失败。
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md:83]:
```suggestion
在这种写入方式下不能手动指定 tag，所有 label 都被视为 tag 并且类型为字符串。请不要尝试使用 SQL 提前创建表来指定 tag 列，这会导致类型不匹配而写入失败。
```

**Review by nicecui — COMMENTED** [review]
please also modify the `overview.md` in the zh docs

**Review by nicecui — APPROVED** [review]

File: i18n/zh/docusaurus-plugin-content-docs/version-0.10/user-guide/ingest-data/for-observerbility/loki.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        "from" = "alloy"
+    }
+}
+```
+
+我们监听文件 `/tmp/foo.txt` 并将日志发送到 GreptimeDB。日志存储在表 `loki_demo_logs` 中，并带有 label `job` 和 `from`。
+
+更多信息，请参考 [Grafana Alloy loki.write 文档](https://grafana.com/docs/alloy/latest/reference/components/loki/loki.write/)。
+
+您可以运行以下命令来检查表中的数据：
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/version-0.10/user-guide/ingest-data/for-observerbility/loki.md:50]:
```suggestion
你可以运行以下命令来检查表中的数据：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        "from" = "alloy"
+    }
+}
+```
+
+我们监听文件 `/tmp/foo.txt` 并将日志发送到 GreptimeDB。日志存储在表 `loki_demo_logs` 中，并带有 label `job` 和 `from`。
+
+更多信息，请参考 [Grafana Alloy loki.write 文档](https://grafana.com/docs/alloy/latest/reference/components/loki/loki.write/)。
+
+您可以运行以下命令来检查表中的数据：
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md:50]:
```suggestion
你可以运行以下命令来检查表中的数据：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+| greptime_timestamp | TimestampNanosecond | PRI  | NO   |         | TIMESTAMP     |
+| line               | String              |      | YES  |         | FIELD         |
++--------------------+---------------------+------+------+---------+---------------+
+5 rows in set (0.00 sec)
+```
+
+- greptime_timestamp: 日志的时间戳。
+- line: 日志消息。
+
+如果您指定了外部 label，我们会将它们添加为表结构中的 tag。例如上面的 `job` 和 `from`。
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md:82]:
```suggestion
如果你指定了外部 label，我们会将它们添加为表结构中的 tag。例如上面的 `job` 和 `from`。
```

File: i18n/zh/docusaurus-plugin-content-docs/version-0.10/user-guide/ingest-data/for-observerbility/loki.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+| greptime_timestamp | TimestampNanosecond | PRI  | NO   |         | TIMESTAMP     |
+| line               | String              |      | YES  |         | FIELD         |
++--------------------+---------------------+------+------+---------+---------------+
+5 rows in set (0.00 sec)
+```
+
+- greptime_timestamp: 日志的时间戳。
+- line: 日志消息。
+
+如果您指定了外部 label，我们会将它们添加为表结构中的 tag。例如上面的 `job` 和 `from`。
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/version-0.10/user-guide/ingest-data/for-observerbility/loki.md:82]:
```suggestion
如果你指定了外部 label，我们会将它们添加为表结构中的 tag。例如上面的 `job` 和 `from`。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+| 2024-11-25 11:02:31.256251 | Greptime is very cool! | /tmp/foo.txt | alloy | greptime |
++----------------------------+------------------------+--------------+-------+----------+
+1 row in set (0.01 sec)
+```
+
+## 数据模型
+
+Loki 日志数据模型根据以下规则映射到 GreptimeDB 数据模型：
+
+没有标签的默认表结构：
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/ingest-data/for-observerbility/loki.md:66]: 标签 -> tag or label?

**Review by fengjiachun — APPROVED** [review]
Rest LGTM

**Pull request merged**